snuba/web/query.py

```diff
@@ -165,8 +167,36 @@ def _format_storage_query_and_run(
     """
 
     source = clickhouse_query.get_data_source().format_from()
-    with sentry_sdk.start_span(description="create_query", op="db"):
-        formatted_query = DictSqlQuery(dataset, clickhouse_query, request_settings)
+    with sentry_sdk.start_span(description="create_query", op="db") as span:
+        legacy_query = DictSqlQuery(dataset, clickhouse_query, request_settings)
+        span.set_data("legacy_query", legacy_query.sql_data())
+        try:
+            ast_query: Optional[AstSqlQuery] = AstSqlQuery(
+                clickhouse_query, request_settings
+            )
+            # For mypy
+            assert ast_query is not None
+            span.set_data("ast_query", ast_query.sql_data())
+
+        except Exception:
+            logger.warning("Failed to format ast query", exc_info=True)
+            ast_query = None
+
+        if (
+            is_ast_rolled_out(get_dataset_name(dataset), referrer)
+            and ast_query is not None
+        ):
+            formatted_query: SqlQuery = ast_query
+            query_type = "ast"
+            metric = "execute.ast"
+        else:
+            formatted_query = legacy_query
+            query_type = "legacy"
+            metric = "execute.legacy"
+
+        metrics.increment(metric)
```

Comment:
Could also be `metrics.increment("execute", tags={"query_type": query_type}` if you want

Reply:
done